ports/stm32/boards/NUCLEO_WB55/rfcore_firmware.py

```diff
@@ -0,0 +1,540 @@
+
+# Argument to `stm.rfcore_fw_version()`.
+_FW_VERSION_FUS = const(0)
+_FW_VERSION_WS = const(1)
+
+# No firmware update in progress. Boot normally.
+_STATE_IDLE = const(0)
+
+# A previous firmware update failed. Will return reason code from resume().
+_STATE_FAILED = const(1)
+
+# Trying to get into the FUS. Keep issuing GET_STATE until the FUS is active.
+_STATE_WAITING_FOR_FUS = const(2)
+
+# Trying to get into the WS. Keep issuing START_WS until the WS is active (or fails).
+_STATE_WAITING_FOR_WS = const(3)
+
+# FW_DELETE has been issued. Waiting for the WS version to report zero.
+_STATE_DELETING_WS = const(4)
+
+# Flash copy has started for FUS/WS. If a reboot occurs, then fail.
+_STATE_COPYING_FUS = const(5)
+_STATE_COPYING_WS = const(6)
+
+# Flash write fully completed, ready for install.
+_STATE_COPIED_FUS = const(7)
+_STATE_COPIED_WS = const(8)
+
+# Check for next update to perform.
+# Either we've just gotten into the FUS, or the first update in a sequence
+# has completed. (e.g. FUS done, now do WS).
+_STATE_CHECK_UPDATES = const(9)
+
+# Installation has started, keep polling GET_STATE.
+_STATE_INSTALLING_WS = const(10)
+_STATE_INSTALLING_FUS = const(11)
+
+# Update completed successfully.
+REASON_OK = 0
+# The device reset during flash copy. Possibly WS still installed.
+REASON_FLASH_COPY_FAILED = 1
+# Unable to start the WS after firmware update.
+REASON_NO_WS = 2
+# Copying FUS image to staging area caused FUS to fail.
+REASON_FLASH_FUS_BAD_STATE = 3
+# Copying WS image to staging area caused FUS to fail.
+REASON_FLASH_WS_BAD_STATE = 4
+# Cannot get into the FUS. Perhaps rfcore misconfigured.
+REASON_FUS_NOT_RESPONDING = 5
+# After a FUS install, unable to get back to the FUS.
+REASON_FUS_NOT_RESPONDING_AFTER_FUS = 6
+# After a WS install, unable to get back to the FUS.
+REASON_FUS_NOT_RESPONDING_AFTER_WS = 7
+# Unable to query rfcore version/active.
+REASON_RFCORE_NOT_CONFIGURED = 8
+# The WS deletion didn't have any effect.
+REASON_WS_STILL_PRESENT = 9
+# FUS refused to delete the WS.
+REASON_WS_DELETION_FAILED = 10
+# FUS returned a specific code for a FUS update.
+# See AN5185 Rev 4, Table 12. Reason between 0x00-0x11 will be added.
+REASON_FUS_VENDOR = 0x10
+# FUS returned a specific code for a WS update. Values as for the FUS update.
+REASON_WS_VENDOR = 0x30
+
+# FUS 1.0.2 must be installed before FUS 1.1.0 can be installed.
+# A factory Nucleo board has FUS (0, 5, 3, 0, 0) and WS (0, 5, 1, 0, 0).
+_FUS_VERSION_102 = (1, 0, 2, 0, 0)
+_FUS_VERSION_110 = (1, 1, 0, 0, 0)
+_PATH_FUS_102 = "fus_102.bin"
+_PATH_FUS_110 = "fus_110.bin"
+_PATH_WS_BLE_HCI = "ws_ble_hci.bin"
+
+# This address is correct for versions up to v1.8 (assuming existing firmware deleted).
+# Note any address from the end of the filesystem to the SFSA would be fine, but if
+# the FUS is fixed in the future to use the specified address then these are the "correct"
+# ones.
+_ADDR_FUS = 0x080EC000
+_ADDR_WS_BLE_HCI = 0x080DC000
```

Comment:
I'm not sure... my thought is that if you're changing this then it's because something has changed about the update procedure there's very likely other changes you need to make.

Reply:
ok